circularBuffer.hpp

```diff
@@ -0,0 +1,116 @@
+#ifndef circularBuffer_
+#define circsularBuffer_
+
+#define BufferSize 5             //circular buffer size 
+
+//**********************************************************************************
+// Author: Sascha Heinisch
+// Date:   28.11.2019
+// 
+// Class: Declarartion of circular buffer class. 
+//
+//**********************************************************************************
+template <class T> class circularBuffer
+{
+  public:
+
+    static T buffer[BufferSize];
```

Comment:
Why static fields?

Reply:
You wrote "Static memory only, no malloc()/new.".

Reply:
Cause you wrote "4. Static memory only, no malloc()/new."

Reply:
Anyway fields don't need to be static to be setup in data memory area. If they are not static but the objects are declared globally they are also stored in data memory.

Reply:
Should I change?

Reply:
No it's not necessary to change, thank you